jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientStreamTest.java

```diff
@@ -1099,6 +1104,61 @@ public boolean process(Request request, org.eclipse.jetty.server.Response respon
         assertTrue(result.isSucceeded());
     }
 
+    @ParameterizedTest
+    @MethodSource("transports")
+    public void testMultiplexedConnectionsForcibleStop(Transport transport) throws Exception
+    {
+        Assumptions.assumeTrue(transport.isMultiplexed());
+
+        long timeoutInSeconds = 5;
+        AtomicInteger processCount = new AtomicInteger();
+        CountDownLatch processLatch = new CountDownLatch(1);
+        startServer(transport, new Handler.Abstract()
+        {
+            @Override
+            public boolean process(Request request, org.eclipse.jetty.server.Response response, Callback callback) throws Exception
+            {
+                processCount.incrementAndGet();
+                processLatch.await(timeoutInSeconds * 2, TimeUnit.SECONDS);
+                callback.succeeded();
+                return true;
+            }
+        });
+        startClient(transport);
+        // Set up the client such as:
+        // - only one connection can be created;
+        // - the connection can be used by two concurrent requests;
+        // - the connection is pre-created.
+        client.setMaxConnectionsPerDestination(1);
+        client.getTransport().setConnectionPoolFactory(destination ->
+        {
+            MultiplexConnectionPool pool = new MultiplexConnectionPool(destination, 1, 2);
+            LifeCycle.start(pool);
+            try
+            {
+                pool.preCreateConnections(1).get();
+            }
+            catch (InterruptedException | ExecutionException e)
+            {
+                throw new RuntimeException(e);
+            }
+            return pool;
+        });
+
+        // Send two parallel requests.
+        client.newRequest(newURI(transport)).send(result -> {});
+        client.newRequest(newURI(transport)).send(result -> {});
+
+        // Wait until both requests are in-flight.
+        await().atMost(timeoutInSeconds, TimeUnit.SECONDS).until(processCount::get, is(2));
+
+        // Stop the client while it has requests in-flight.
+        Assertions.assertTimeout(Duration.ofSeconds(timeoutInSeconds), () -> LifeCycle.stop(client));
+
+        // Let the server threads go.
+        processLatch.countDown();
```

Comment:
Probably a good idea here to wait for a latch that is counted down in `send(result -> ...)`, otherwise we risk to stop the server while still inside the `Handler`, so it'll throw exceptions, etc.

Reply:
Done.